src/aligners/tests.rs

```diff
@@ -383,4 +383,183 @@ mod dt {
             }],
         ));
     }
+    //HomoPolymer tests
+
+    #[test]
+    fn ins_homopolymer_cost() {
+        test(AffineCost::new(
+            Some(1),
+            Some(1),
+            None,
+            [AffineLayerCosts {
+                affine_type: AffineLayerType::HomoPolymerInsert,
+                open: 2,
+                extend: 2,
+            }],
+        ));
+    }
+
+    #[test]
+    fn del_homopolymer_cost() {
+        test(AffineCost::new(
+            Some(1),
+            None,
+            Some(1),
+            [AffineLayerCosts {
+                affine_type: AffineLayerType::HomoPolymerDelete,
+                open: 2,
+                extend: 2,
+            }],
+        ));
+    }
+
+    #[test]
+    fn indel_homopolymer_plus_affine_cost() {
+        test(AffineCost::new(
+            Some(1),
+            Some(1),
```

Comment:
Disallow linear `ins` and `del` here by making them `None`, since otherwise the affine layers won't be used at all